src/parse.y

```diff
@@ -1216,16 +1216,42 @@ func_body:
 	;
 
 anonymous_function:
-		TOK_FUNCTION begin_func func_body
-			{ $$ = new ConstExpr($2->ID_Val()); }
+		TOK_FUNCTION func_params
+			{
+			$<id>$ = current_scope()->GenerateTemporary("lambda-function");
+			begin_func($<id>$, current_module.c_str(), FUNC_FLAVOR_FUNCTION, 0, $2);
+			}
```

Comment:
This could call `begin_func` here. I decided not to though because unless we decide to make all anonymous functions capture their scope (specifically ones the are the default for tables) calling these functions `lambda-function` adds some clarity at the cost of two lines of mostly redundant code.

Reply:
Agree with Jon here that we should cover all cases for consistency.